pubsublite/internal/wire/streams.go

```diff
@@ -88,10 +96,11 @@ type streamHandler interface {
 // are private implementation.
 type retryableStream struct {
 	// Immutable after creation.
-	ctx          context.Context
-	handler      streamHandler
-	responseType reflect.Type
-	timeout      time.Duration
+	ctx               context.Context
+	handler           streamHandler
+	responseType      reflect.Type
+	timeout           time.Duration
+	streamInitTimeout time.Duration
```

Comment:
I'm really sorry to slow you down, but would you mind splitting out the changes for error handling from those for the stream initialization retry? I think the former are a must have, but I am slightly concerned about the latter (I'm sorry I didn't raise this before, we talked about this already).

I'm a little worried about the complexity, and the lack of consistency across clients. Can we discuss it during our dev sync today?

Reply:
Done. I've removed stream init retries from this PR.